File: CPlusPlus/388.LongestAbsoluteFilePath.cpp
class Solution {
public:
    int lengthLongestPath(string input) {
        int count = 0;
        int maxLen = 0;
        int currLevel = 1;
        vector<int> level(200);
        int n = input.size();
        bool isFile = false;
        for (int i = 0; i < n; ++i) {
            while (input[i] == '\t' && i < n) {
                ++currLevel; ++i;
            }
            while (input[i] != '\n' && i < n) {
                if (input[i] == '.') isFile = true;
                ++count; ++i;
            }
            if (isFile) {
                maxLen = max(maxLen, level[currLevel-1] + count);
            } else {
                // + 1 means '/'
                level[currLevel] = level[currLevel-1] + count + 1;
            }
            // reset
            count = 0; isFile = false; currLevel = 1;
        }
        return maxLen;
    }
};

File: Javascript/388.LongestAbsoluteFilePath.js
/**
 * @param {string} input
 * @return {number}
 */
var lengthLongestPath = function(input) {
  var ret = 0;
  var currPathLength = 0;
  var levelLength = [0];
  var i = 0;
  var level = 0;
  var isFile = false;
  while (i < input.length) {
    if (input[i] == '\n') {
        levelLength[level+1] = currPathLength;
        level = 0;
        currPathLength = 0;
        i++;
        isFile = 0;
    } else if (input[i] == '\t') {
      level++;
      currPathLength = levelLength[level]+1;
      i++;
    } else if (input[i] == '.') {
      isFile = true;
      currPathLength++;
      i++;
    } else {
      currPathLength++;
      if (isFile)
        ret = Math.max(ret, currPathLength)
      i++;
    }
  }
  // console.log(levelLength)
  return ret;
};